src/main/java/picocli/CommandLine.java

```diff
@@ -1929,10 +1929,30 @@ public int execute(String... args) {
         }
     }
     private static int handleUnhandled(Exception ex, CommandLine cmd, int defaultExitCode) {
-        ex.printStackTrace(cmd.getErr());
+        cmd.getErr().print(exceptionToColorString(ex, cmd.getColorScheme()));
+        cmd.getErr().flush();
         return mappedExitCode(ex, cmd.getExitCodeExceptionMapper(), defaultExitCode);
     }
 
+    /**
+     * Convert an {@code Exception} to a {@code String} , with message and stack traces extracted and colored
+     * according to {@code ColorScheme}.
+     * @param ex the {@code Exception} to be converted
+     * @param colorScheme the {@code ColorScheme} to use
+     * @return converted and colored {@code String}
+     */
+    private static String exceptionToColorString(Exception ex, Help.ColorScheme existingColorScheme){
+        Help.ColorScheme colorScheme = new Help.ColorScheme.Builder(existingColorScheme).applySystemProperties().build();
```

Comment:
Out of interest: why not use the specified `existingColorScheme` directly? Is there a reason to make a copy?

Reply:
Sorry... Just noticed this after a few days. One reason I'm using a copy here is not sure if the system property about the color setting might be changing during runtime, and another reason is this is how `Help` constructor works, so I'd like to comply with what's existing here...